kapten/github.py

```diff
@@ -0,0 +1,115 @@
+        return False
+    except Exception as e:  # pragma: nocover
+        logger.critical(e)
+        return False
+
+
+def parse_webhook_payload(
+    payload: Dict[str, Any], tracked_repositories: List[str]
+) -> Tuple[str, str]:
+    # Validate payload structure
+    valid_structure = payload and all(
+        [
+            "deployment" in payload,
+            "statuses_url" in payload.get("deployment", {}),
+            "payload" in payload.get("deployment", {}),
+            "digest" in payload.get("deployment", {}).get("payload", {}),
+            "tag" in payload.get("deployment", {}).get("payload", {}),
+            "image" in payload.get("deployment", {}).get("payload", {}),
+            "repository" in payload,
+            "full_name" in payload.get("repository", {}),
+        ]
+    )
+    if not valid_structure:
+        raise ValueError("Invalid GitHub payload")
+
+    deployment_payload = json.loads(payload["deployment"]["payload"])
+    image = deployment_payload["image"]
+    if image not in tracked_repositories:
+        raise ValueError("No image value")
+
+    callback_url = payload["deployment"]["statuses_url"]
+    if not callback_url.startswith("https://api.github.com/repos/"):
+        raise ValueError("Invalid GitHub callback url: {}".format(callback_url))
+
+    tag = deployment_payload["tag"]
+    if not tag:
+        raise ValueError("No tag value")
+
+    # Format: <IMAGE>@<DIGEST> where DIGEST is prefixed with: 'sha256' expected
+    digest = deployment_payload["digest"]
+    if "@" not in digest:
+        raise ValueError("Invalid digest value: {}".format(digest))
+    digest = digest.split("@")[1]
+    if not digest.startswith("sha256:"):
+        raise ValueError("Invalid digest value: {}".format(digest))
+
+    return f"{image}:{tag}@{digest}", callback_url
+
+
+async def callback(url: str, state: str, environment: str, description: str) -> bool:
+    # TODO: Also accept: 'log_url', 'environment_url'
+    # TODO: Authentication (token?) for GitHub API
+    valid_states = {
+        "error",
+        "failure",
+        "inactive",
+        "in_progress",
+        "queued",
+        "pending",
+        "success",
+    }
+    if state not in valid_states:
+        raise ValueError("Invalid state: {}".format(state))
+
+    data = {
+        "state": state,
+        "description": description,
+        "environment": environment,
+    }
+    async with httpx.Client() as client:
+        # TODO: Add the special media types in Accept header
+        #       See: https://developer.github.com/v3/repos/deployments/#create-a-deployment-status
+        try:
+            response = await client.request("POST", url, json=data)
+        except Exception as e:  # pragma: nocover
+            # TODO: Retry
+            logger.critical(e)
+            return False
+
+        if response.is_error:
+            error = response.json()
+            logger.critical("Error response from GitHub: {}".format(error))
```

Comment:
Log `repr` of the json might be more safe here. And also not pre-format.

```suggestion
            logger.critical("Error response from GitHub: %r", error)
```